prime-router/src/main/kotlin/fhirengine/translation/hl7/utils/CustomFHIRFunctions.kt

```diff
@@ -1,22 +1,40 @@
 package gov.cdc.prime.router.fhirengine.translation.hl7.utils
 
 import ca.uhn.fhir.model.api.TemporalPrecisionEnum
-import gov.cdc.prime.router.Metadata
 import gov.cdc.prime.router.fhirengine.translation.hl7.SchemaException
-import gov.cdc.prime.router.metadata.LivdLookup
 import org.hl7.fhir.r4.model.Base
 import org.hl7.fhir.r4.model.BaseDateTimeType
 import org.hl7.fhir.r4.model.BooleanType
 import org.hl7.fhir.r4.model.DateTimeType
-import org.hl7.fhir.r4.model.Device
 import org.hl7.fhir.r4.model.IntegerType
-import org.hl7.fhir.r4.model.Observation
 import org.hl7.fhir.r4.model.StringType
 import org.hl7.fhir.r4.utils.FHIRPathEngine
 import java.time.DateTimeException
 import java.time.ZoneId
 import java.util.TimeZone
 
+/**
+ * This interface contains the required method signatures required to implement custom FHIR functions
+ */
+interface FhirPathFunctions {
+
+    /**
+     * Get the function details for a given [functionName].
+     * @return the function details
+     */
+    fun resolveFunction(functionName: String?): FHIRPathEngine.IEvaluationContext.FunctionDetails?
+
+    /**
+     * Execute the function on a [focus] resource for a given [functionName] and [parameters].
+     * @return the function result
+     */
+    fun executeFunction(
+        focus: MutableList<Base>?,
+        functionName: String?,
+        parameters: MutableList<MutableList<Base>>?
+    ): MutableList<Base>
+}
+
 /**
  * Custom FHIR functions created by report stream to help map from FHIR -> HL7
  * only used in cases when the same logic couldn't be accomplished using the FHIRPath
```

Comment:
It seems like it would be cleaner for this to be a class that implements the interface?

Reply:
Good idea, added that.